### DISCUSSION TITLE
Is nodata applied before expressions?

### DISCUSSION THREAD

**Discussion by agasparovic** [developmentseed/titiler#976] [Q&A] [answered]
Hi there,

I have a band with values from 0.0 to 1.0. I'm attempting to make any values below 0.15 transparent by doing the following:

* Setting `nodata=0.0`
* Using an expression like: `where(b15>=0.15, b15, 0.0)` (b15 is the band)

It seems to apply the expression correctly, but makes the areas where b15 is < 0.15 black (the lowest color in the colormap) instead of transparent. I'm guessing nodata is evaluated before the expression maybe?

Is there something else I could be doing to make this work?

Thanks!
Andrew

**Comment by vincentsarago** [comment, marked as answer]
Hi @agasparovic 

Thanks for starting this discussion.

The mask is created for the data itself, expression won't update the mask.

Can you tell a bit more of what you're doing: what rescaling or colormap you are using? 

> Is there something else I could be doing to make this work?

Custom Algorithm might solve this for you (https://developmentseed.org/titiler/advanced/Algorithms/, https://developmentseed.org/titiler/examples/code/tiler_with_custom_algorithm/) 

👇 not tested so might be wrong 
```python
import numpy
from titiler.core.algorithm import BaseAlgorithm
from rio_tiler.models import ImageData

class Correct(BaseAlgorithm):

    input_nbands: int = 1
    output_nbands: int = 1

    def __call__(self, img: ImageData) -> ImageData:
        new_mask = numpy.where(im.array >= 0.15, False, True)
        img.array.mask = numpy.bitwise_or(new_mask, self.mask)
        return img
```

**Reply by agasparovic** [reply]
Thank you Vincent! A custom algorithm works great.
